src/codegeneration/ES6ClassTransformer.js

```diff
@@ -0,0 +1,265 @@
+// limitations under the License.
+
+import {
+  CONSTRUCTOR
+} from '../syntax/PredefinedName.js';
+import {
+  AnonBlock,
+  ClassExpression,
+  ClassDeclaration,
+  GetAccessor,
+  PropertyMethodAssignment,
+  SetAccessor,
+  LiteralPropertyName,
+} from '../syntax/trees/ParseTrees.js';
+import {
+  CALL_EXPRESSION,
+  EXPRESSION_STATEMENT,
+  GET_ACCESSOR,
+  PROPERTY_METHOD_ASSIGNMENT,
+  PROPERTY_VARIABLE_DECLARATION,
+  SET_ACCESSOR,
+  SUPER_EXPRESSION
+} from '../syntax/trees/ParseTreeType.js';
+import {TempVarTransformer} from './TempVarTransformer.js';
+import {ParseTreeTransformer} from './ParseTreeTransformer.js';
+import {
+  createBindingIdentifier,
+  createCommaExpression,
+  createEmptyParameterList,
+  createExpressionStatement,
+  createFunctionBody,
+  createIdentifierToken,
+  createImmediatelyInvokedFunctionExpression,
+  createPropertyNameAssignment,
+  createParenExpression,
+} from './ParseTreeFactory.js';
+import {
+  parseExpression,
+  parsePropertyDefinition,
+  parseStatement,
+} from './PlaceholderParser.js';
+import {propName} from '../staticsemantics/PropName.js';
+import {prependStatements} from './PrependStatements.js';
+
+// TODO(vicb): comment
+
+export class ES6ClassTransformer extends TempVarTransformer {
+  /**
+   * @param {UniqueIdentifierGenerator} identifierGenerator
+   */
+  constructor(identifierGenerator) {
+    super(identifierGenerator);
+  }
+
+  transformClassElements_(tree) {
+    let superClass = this.transformAny(tree.superClass);
+
+    let elements = [];
+    let initInstanceVars = [], initStaticVars = [];
+    let constructor;
+    let constructorIndex = 0;
+
+    tree.elements.forEach((tree) => {
+      let initVars;
+      if (tree.isStatic) {
+        initVars = initStaticVars;
+      } else {
+        initVars = initInstanceVars;
+      }
+
+      switch (tree.type) {
+        case GET_ACCESSOR:
+          let getter = new GetAccessor(tree.location, tree.isStatic, tree.name,
+              tree.typeAnnotation, tree.annotations, tree.body);
+          elements.push(this.transformAny(getter));
+          break;
+
+        case SET_ACCESSOR:
+          let setter = new SetAccessor(tree.location, tree.isStatic, tree.name,
+              tree.parameterList, tree.annotations, tree.body);
+          elements.push(this.transformAny(setter));
+          break;
+
+        case PROPERTY_METHOD_ASSIGNMENT:
+          if (!tree.isStatic && propName(tree) === CONSTRUCTOR) {
+            constructor = tree;
+            constructorIndex = elements.length;
+          } else {
+            let pma = new PropertyMethodAssignment(tree.location, tree.isStatic,
+                tree.functionKind, tree.name, tree.parameterList,
+                tree.typeAnnotation, tree.annotations, tree.body,
+                tree.debugName);
+            elements.push(this.transformAny(pma));
+          }
+          break;
+
+        case PROPERTY_VARIABLE_DECLARATION:
+          tree = this.transformAny(tree);
```

Comment:
add a transformPropertyVariableDeclaration that handles this transformatio instead of using getInstanceInitExpression.

```js
transformPropertyVariableDeclaration(tree) {
  return new BinaryExpression(tree.location, 
      new MemberExpression(...),
      EQUAL,
      this.transformAny(tree.initializer))
}
```

// pseudo code